src/utils/batched-updates.js

```diff
@@ -8,23 +8,33 @@ import {
 import defaults from '../defaults';
 import supports from './supported-features';
 
-let isInsideBatchedSchedule = false;
+let batchedScheduled = false;
+
+let batched = [];
+
+const executeBatched = () => {
+  unstable_batchedUpdates(() => {
+    while (batched.length) {
+      const currentBatched = batched;
+      batched = [];
+      currentBatched.forEach((fn) => fn());
+    }
+    // important to reset it before exiting this function
+    // as React will dump everything right after
+    batchedScheduled = false;
```

Comment:
- all events will be executed in the "actually correct order" - First in, First out.
- all will be executed in one batch, mimicking React 18 behavior and reducing re-renders

Reply:
Let's maybe put this in a minor, but worth calling it out. Also eager to see the effect on Jira and how many things break there 🙈

Reply:
Let's maybe put this in a minor, but eager to test in Jira and see what breaks there 🙈

Reply:
It breaks one component doing read right after set in useEffect (`QueryRenderer` component). It's not clear how it was able to read the data also synchronously before.

Removing `scheduleCallback` fixes the problem and I think we can/need to remove it for React 18 auto-batching.
It's removal also might help with #219 as "flush" will be performed early.